esp8266/modesp.c

```diff
@@ -35,6 +35,492 @@
 #include MICROPY_HAL_H
 #include "queue.h"
 #include "user_interface.h"
+#include "espconn.h"
+
+#include "utils.h"
+
+#define IPV4ADDR_BUFSIZE    4
+
+// Helpers from cc3200/mods/modnetwork.c
+STATIC void esp_espconn_parse_ipv4_addr(mp_obj_t addr_in, uint8_t *out_ip) {
```

Comment:
We should try our best not to duplicate code across different ports, and this code already appears in cc3200 and stmhal separately.  Is your code an exact copy from cc3200?  If so we should factor it into a separate helper file common to all ports.